core/src/main/java/org/apache/gravitino/authorization/OwnerManager.java

```diff
@@ -0,0 +1,193 @@
+      LOG.error(errorMsg);
+      throw new RuntimeException(errorMsg);
+    }
+    this.store = store;
+  }
+
+  public void setOwner(
+      String metalake, MetadataObject metadataObject, String ownerName, Owner.Type ownerType) {
+    try {
+      NameIdentifier objectIdent = MetadataObjectUtil.toEntityIdent(metalake, metadataObject);
+      if (ownerType == Owner.Type.USER) {
+        NameIdentifier ownerIdent = AuthorizationUtils.ofUser(metalake, ownerName);
+        TreeLockUtils.doWithTreeLock(
+            objectIdent,
+            LockType.READ,
+            () ->
+                TreeLockUtils.doWithTreeLock(
+                    ownerIdent,
+                    LockType.READ,
+                    () -> {
+                      store
+                          .relationOperations()
+                          .insertRelation(
+                              SupportsRelationOperations.Type.OWNER_REL,
+                              objectIdent,
+                              MetadataObjectUtil.toEntityType(metadataObject),
+                              ownerIdent,
+                              Entity.EntityType.USER,
+                              true);
+                      return null;
+                    }));
+      } else if (ownerType == Owner.Type.GROUP) {
+        NameIdentifier ownerIdent = AuthorizationUtils.ofGroup(metalake, ownerName);
+        TreeLockUtils.doWithTreeLock(
+            objectIdent,
+            LockType.READ,
+            () ->
+                TreeLockUtils.doWithTreeLock(
+                    ownerIdent,
+                    LockType.READ,
+                    () -> {
+                      store
+                          .relationOperations()
+                          .insertRelation(
+                              SupportsRelationOperations.Type.OWNER_REL,
+                              objectIdent,
+                              MetadataObjectUtil.toEntityType(metadataObject),
+                              ownerIdent,
+                              Entity.EntityType.GROUP,
+                              true);
+                      return null;
+                    }));
+      }
+    } catch (NoSuchEntityException nse) {
+      LOG.warn(
+          "Metadata object {} or owner {} is not found", metadataObject.fullName(), ownerName, nse);
+      throw new NotFoundException(
+          nse, "Metadata object %s or owner %s is not found", metadataObject.fullName(), ownerName);
+    } catch (IOException ioe) {
+      LOG.info(
+          "Fail to set the owner {} of metadata object {}",
+          ownerName,
+          metadataObject.fullName(),
+          ioe);
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  public Optional<Owner> getOwner(String metalake, MetadataObject metadataObject) {
+    try {
+      OwnerImpl owner = new OwnerImpl();
+      NameIdentifier ident = MetadataObjectUtil.toEntityIdent(metalake, metadataObject);
+      List entities =
```

Comment:
Why do we use List without type, I think it has compile warning.

Reply:
If I use entity here, it will throw an error.

Reply:
Is that ok?
```
List<? extends Entity> entities = 
```

Reply:
OK, It can compile.